test/DynamicExpresso.UnitTest/ConstructorTest.cs

```diff
@@ -179,6 +180,148 @@ public void Array_multi_dimension_constructor()
+			var args = new object[]
+			{
+				strProp.Value
+			};
+			Assert.AreEqual(
+				new MyClassAdder() { { 1, 2, 3, 4, 5 }, "6", 7 },
+				target.Parse("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7}", strProp).Invoke(args));
+			Assert.AreEqual(
+				new MyClassAdder() { { 1, 2, 3, 4, 5 }, string.Empty, 7 },
+				target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},string.Empty, 7}"));
+			Assert.AreEqual(
+				new MyClassAdder() { StrProp = string.Empty, MyArr = new[] { 1, 2, 3, 4, 5 }, IntField = int.MinValue },
+				target.Eval<MyClassAdder>("new MyClassAdder() {StrProp = string.Empty, MyArr = new int[] {1, 2, 3, 4, 5}, IntField = int.MinValue }"));
+		}
+
+		[Test]
+		public void Ctor_NewMyClass_CanStillUseMemberSyntax()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.AreEqual(
+				new MyClassAdder() { StrProp = string.Empty, MyArr = new[] { 1, 2, 3, 4, 5 }, IntField = int.MinValue },
+				target.Eval<MyClassAdder>("new MyClassAdder() {StrProp = string.Empty, MyArr = new int[] {1, 2, 3, 4, 5}, IntField = int.MinValue }"));
+		}
+
+		[Test]
+		public void Ctor_InvalidInitializerMemberDeclarator()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7	}"));
+			//Start with collection, then do member init
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},StrProp = \"6\" ,7	}"));
+			//Member init first, then attempt a collection init.
+			Assert.Throws<ParseException>(() => target.Eval<MyClassAdder>("new MyClassAdder(){StrProp = \"6\" ,{ 1, 2, 3, 4, 5},7	}"));
+		}
+
+		[Test]
+		public void Ctor_CannotUseCollectionInitDoesNotImplementIEnumerable()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(MyClassAdder));
+			target.Reference(typeof(MyClass));
+			Assert.Throws<ParseException>(() => target.Eval<MyClass>("new MyClass(){ 1, 2, 3, 4, 5}"));
+		}
+
+		[Test]
+		public void Ctor_NewListWithItems()
+		{
+			Ctor_NewListGeneric<string, string>("\"1\"", "\"2\"", "\"3\"", "{string.Empty}", "string.Empty", "int.MaxValue.ToString()", "{int.MinValue.ToString()}");
+			Ctor_NewListGeneric<int, int>("1", "2", "3", "int.MinValue", "int.MaxValue", "{int.MinValue}", "{int.MaxValue}");
+			Ctor_NewListGeneric<object, object>("string.Empty", "int.MinValue");
+		}
+
+		[Test]
+		public void Ctor_NewListCantFindAddMethod()
+		{
+			var target = new Interpreter();
+			target.Reference(typeof(System.Collections.Generic.List<>));
+			Assert.Throws<ParseException>(() => target.Eval<System.Collections.Generic.List<int>>("string.Empty"));
+			Assert.Throws<ParseException>(() => target.Eval<System.Collections.Generic.List<string>>("int.MaxValue"));
```

Comment:
I think you're missing the call to the List constructor here.

Reply:
Correct